runner.go

```diff
@@ -227,19 +236,20 @@ func NewRunner(p RunnerParams) *Runner {
 // when the worker exits, an AlreadyExists error will be returned.
 //
 // StartWorker returns ErrDead if the runner is not running.
-func (runner *Runner) StartWorker(id string, startFunc func() (Worker, error)) error {
+func (runner *Runner) StartWorker(ctx context.Context, id string, startFunc StartFunc) error {
```

Comment:
So this part *is* an API breakage, we might as well get whatever we want from the rest of it while we are adding context and updating all the callers.
*Is* it cleaner to reflect for this information, or is it more obvious to also pass in a string (label/name/etc) at this point)

Reply:
Yeah, this is why this is in `master` branch, which is officially the `v4` (once beta2 of juju/juju is released, juju will switch to v4). So I'm happy to make these changes here.

`context.Context` should probably be passed here anyway, so the question becomes, should we pass the labels on the context or explicitly pass them.

I'll think about this one.

Reply:
I've removed the reflect and instead decided to use a `Name` on the config/params of catacomb and runner. I think this will give better deterministic results that can be identified in profiling applications.